src/App.js

```diff
@@ -14,11 +15,118 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goodsList: [...goodsFromServer],
+    appStarted: false,
+    isReverse: false,
+    sorting: 'default',
+  }
+
+  setVisible = () => {
+    this.setState(state => ({
+      appStarted: !state.appStarted,
+    }));
+  }
+
+  makeReverse = () => {
+    this.setState(state => ({
+      isReverse: !state.isReverse,
+    }));
+  }
+
+  alphabetSort = () => {
+    this.setState({ sorting: 'alphabet' });
+  }
+
+  lengthSort = () => {
+    this.setState({ sorting: 'length' });
+  }
+
+  reset = () => {
+    this.setState({
+      goodsList: [...goodsFromServer],
+      sorting: 'default',
+      isReverse: false,
+    });
+  }
+
+  render() {
+    const {
+      appStarted,
+      sorting,
+      isReverse,
+      goodsList,
+    } = this.state;
+
+    goodsList.sort((a, b) => {
+      switch (sorting) {
+        case 'alphabet':
+          return a.localeCompare(b);
+
+        case 'length':
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReverse) {
+      goodsList.reverse();
+    }
```

Comment:
move this out of render method
create function getSortedGoods, which returns a list of goods after manipulating them according to the desired action (reverse, sort alphabetically or sort by length)